fhir-persistence-jdbc/src/main/java/com/ibm/fhir/persistence/jdbc/util/type/NumberParmBehaviorUtil.java

```diff
@@ -0,0 +1,289 @@
+
+/**
+ * Number Parameter Behavior Util encapsulates the logic specific to Prefix
+ * treatment of the
+ * Search queries related to numbers, and the implied ranges therein.
+ */
+public class NumberParmBehaviorUtil {
+    private static final Logger log = java.util.logging.Logger.getLogger(NumberParmBehaviorUtil.class.getName());
+
+    private static final BigDecimal LOWER_BOUND = new BigDecimal(".9");
+    private static final BigDecimal UPPER_BOUND = new BigDecimal("1.1");
+
+    private NumberParmBehaviorUtil() {
+        // No operation
+    }
+
+    public static void executeBehavior(StringBuilder whereClauseSegment, Parameter queryParm,
+            List<Object> bindVariables, Class<?> resourceType, String tableAlias, JDBCQueryBuilder queryBuilder)
+            throws FHIRPersistenceException {
+        // Start the Clause 
+        // Query: AND (
+        whereClauseSegment.append(AND).append(LEFT_PAREN);
+
+        // Process each parameter value in the query parameter
+        boolean parmValueProcessed = false;
+        Set<String> seen = new HashSet<>();
+        for (ParameterValue value : queryParm.getValues()) {
+
+            Prefix prefix = value.getPrefix();
+
+            // Default to EQ
+            if (prefix == null) {
+                prefix = Prefix.EQ;
+            }
+            BigDecimal originalNumber = value.getValueNumber();
+
+            // seen is used to optimize against a repeated value passed in. 
+            // the hash must use the prefix and original number. 
+            String hash = prefix.value() + originalNumber.toPlainString();
+            if (!seen.contains(hash)) {
+                seen.add(hash);
+                // Check for valid conditions and return if it needs to be treated as an INTEGER:
+                boolean isInteger = checkIntegerSearchWithSaEb(prefix, resourceType, queryParm, originalNumber);
+
+                // If multiple values are present, we need to OR them together.
+                if (parmValueProcessed) {
+                    // ) OR (
+                    whereClauseSegment.append(RIGHT_PAREN)
+                            .append(JDBCOperator.OR.value())
+                            .append(LEFT_PAREN);
+                } else {
+                    // Signal to the downstream to treat any subsequent value as an OR condition 
+                    parmValueProcessed = true;
+                }
+
+                // Branch behavior based on the type of prefix. 
+                // EQ/NE/AP have special handling
+                switch (prefix) {
+                case EQ:
+                    // eq - equals is a specific range search.
+                    if (isInteger) {
+                        bindVariables.add(originalNumber);
+                        whereClauseSegment.append(tableAlias + DOT).append(NUMBER_VALUE);
+                        whereClauseSegment.append(JDBCOperator.EQ.value())
+                                .append(BIND_VAR);
```

Comment:
i thought we talked about just handling integer values the same as decimals, but should be fine either way

Reply:
Right.  Sorry, this should be OK - it's heavily tested.  But yes, you are right.